File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_d = {}
        
        for index, num in enumerate(nums):
            if num in nums_d and 2 * num ==  target:
                return [index, nums_d[num]]
                                
            nums_d[num] = index
            
            num_2 = target - num
            if num_2 != num and num_2 in nums_d:
                index_2 = nums_d[num_2]
                
                return [index, index_2]
            

File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;
use std::convert::TryInto;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen: HashMap<i32, i32> = HashMap::new();
        let mut answer: Vec<i32> = Vec::new();
        
        
        for (i, num) in nums.iter().enumerate() {            
            if (seen.contains_key(& (target - num))) {
                answer.push(i.try_into().unwrap());
                answer.push(
                    match seen.get(& (target - num)) {
                        Some(v) => *v,
                        None => *num
                    }
                );
                break;
            }
            
            seen.insert(*num, i.try_into().unwrap());
        }
        
        return answer;
    }
}